### PR TITLE
Basic, automated integration tests on CI

### PR DESCRIPTION
**Pull request by zbigg**
* Add in-browser integration tests.
* Run tests in ChromeHeadless using karma.

Prerequisite for #10 #13 
Related to HARP-7840

### CODE REVIEW

File: test/HarpLeafletBasicTest.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        //await new Promise(resolve => setTimeout(resolve, 10));
+
+        const onAddStub = sandbox.spy(HarpGL.prototype, "onAdd");
+        harpGLLayer = new HarpGL({ theme: {} }).addTo(leafletMap);
+
+        assert.equal(onAddStub.callCount, 1, "onAdd should be called");
+    });
+
+    it("position is set correctly", async function() {
+        await new Promise(resolve => setTimeout(resolve, 20));
```
Comment (ninok) [inline test/HarpLeafletBasicTest.ts]: Why is that?

File: test/HarpLeafletBasicTest.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+        const lZoom = leafletMap.getZoom();
+        const lCenter = leafletMap.getCenter();
+
+        const harpCenter = harpGLLayer.mapView.geoCenter;
+        const harpZoom = harpGLLayer.mapView.zoomLevel;
+
+        const geoEpsilon = 0.00000001;
+        assert.closeTo(harpCenter.latitude, lCenter.lat, geoEpsilon);
+        assert.closeTo(harpCenter.longitude, lCenter.lng, geoEpsilon);
+        assert.closeTo(harpZoom, lZoom, geoEpsilon);
```
Comment (ninok) [inline test/HarpLeafletBasicTest.ts]: Shouldn't the zoomlevel actually match or at least have a different threshold?

File: test/HarpLeafletBasicTest.ts (hunk 1 above)
Comment (zbigg) [reply test/HarpLeafletBasicTest.ts]: Changed to
```
    // `mapView.zoomLevel` is updated only when actual rendering occurs, so we force frame
    // update to ensure that zoom level is applied.
    harpGLLayer.mapView.update();
    await waitForEvent(harpGLLayer.mapView, MapViewEventNames.AfterRender);
```

File: test/HarpLeafletBasicTest.ts (hunk 2 above)
Comment (zbigg) [reply test/HarpLeafletBasicTest.ts]: Fixed

**Pull request merged**